### ISSUE TITLE
Linking a fastapi-users User model with One-To-Many relationship creates error for the select field in Admin Panel

### ISSUE THREAD

**Issue by florianabel** [aminalaee/sqladmin#529]
### Checklist

- [X] The bug is reproducible against the latest release or `master`.
- [X] There are no similar issues or pull requests to fix it yet.

### Describe the bug

Firstof, thanks a lot for the package, it comes in really handy!

Unfortunately I encountered a bug when having my User model linked to other models with a One-To-Many relationship.

I have linked the User model (which is from fastapi-users) to some of my models with a One-To-Many relationship.
When having (at least) one row in the users table and trying to create a new object for another model that is linked to it the query for the select seems to produce an error for a missing unique method.

`sqlalchemy.exc.InvalidRequestError: The unique() method must be invoked on this Result, as it contains results that include joined eager loads against collections`

Based on the stack trace it seems to originate from here:
`File "/usr/local/lib/python3.11/site-packages/sqladmin/forms.py", line 232, in _prepare_select_options`

I am not sure why the error occurs only with the fastapi-users User model but not with other One-To-Many relationships. However, it seems, that maybe adding the unique() method there might solve the problem.

### Steps to reproduce the bug

```
class User(SQLAlchemyBaseUserTable[int], Base):
    id: int = Column(Integer, primary_key=True)
    oauth_accounts: Mapped[List[OAuthAccount]] = relationship(
        "OAuthAccount", lazy="joined"
    )

class Project(Base):
    __tablename__ = "project"

    id: int = Column(Integer, primary_key=True)
    user_id: int = Column(Integer, ForeignKey('user.id'))
    user = relationship("User")
    no_error_id: int = Column(
        Integer, ForeignKey('random_table.id'), nullable=True)
    no_error = relationship("RandomTable", foreign_keys=[no_error_id])
```

Adding the User relationship produces the error, whereas adding another relationship does not.

The error only occurs, when there is already at least one user in the table. As long as the query comes back empty the error is not produced.

I hope I explained the error well enough and am looking forward to your ideas.

Thank you already.

### Expected behavior

I can create a new Object of e.g. "Project" in the Admin Panel without throwing an error

### Actual behavior

When a user object exists and I am trying to create an object of Project in the Admin Panel I receive an Internal Server Error

### Debugging material

appshots-fastapi-myappshots-1  | INFO:     172.28.0.1:60418 - "GET /project/create HTTP/1.1" 500 Internal Server Error
appshots-fastapi-myappshots-1  | ERROR:    Exception in ASGI application
appshots-fastapi-myappshots-1  | Traceback (most recent call last):
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/protocols/http/httptools_impl.py", line 435, in run_asgi
appshots-fastapi-myappshots-1  |     result = await app(  # type: ignore[func-returns-value]
appshots-fastapi-myappshots-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
appshots-fastapi-myappshots-1  |     return await self.app(scope, receive, send)
appshots-fastapi-myappshots-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/applications.py", line 276, in __call__
appshots-fastapi-myappshots-1  |     await super().__call__(scope, receive, send)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/applications.py", line 122, in __call__
appshots-fastapi-myappshots-1  |     await self.middleware_stack(scope, receive, send)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 184, in __call__
appshots-fastapi-myappshots-1  |     raise exc
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 162, in __call__
appshots-fastapi-myappshots-1  |     await self.app(scope, receive, _send)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/sessions.py", line 86, in __call__
appshots-fastapi-myappshots-1  |     await self.app(scope, receive, send_wrapper)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/cors.py", line 83, in __call__
appshots-fastapi-myappshots-1  |     await self.app(scope, receive, send)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
appshots-fastapi-myappshots-1  |     raise exc
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
appshots-fastapi-myappshots-1  |     await self.app(scope, receive, sender)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
appshots-fastapi-myappshots-1  |     raise e
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
appshots-fastapi-myappshots-1  |     await self.app(scope, receive, send)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 718, in __call__
appshots-fastapi-myappshots-1  |     await route.handle(scope, receive, send)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 443, in handle
appshots-fastapi-myappshots-1  |     await self.app(scope, receive, send)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/applications.py", line 122, in __call__
appshots-fastapi-myappshots-1  |     await self.middleware_stack(scope, receive, send)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 184, in __call__
appshots-fastapi-myappshots-1  |     raise exc
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 162, in __call__
appshots-fastapi-myappshots-1  |     await self.app(scope, receive, _send)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
appshots-fastapi-myappshots-1  |     raise exc
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
appshots-fastapi-myappshots-1  |     await self.app(scope, receive, sender)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 718, in __call__
appshots-fastapi-myappshots-1  |     await route.handle(scope, receive, send)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 276, in handle
appshots-fastapi-myappshots-1  |     await self.app(scope, receive, send)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 66, in app
appshots-fastapi-myappshots-1  |     response = await func(request)
appshots-fastapi-myappshots-1  |                ^^^^^^^^^^^^^^^^^^^
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/sqladmin/authentication.py", line 62, in wrapper_decorator
appshots-fastapi-myappshots-1  |     return await func(*args, **kwargs)
appshots-fastapi-myappshots-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/sqladmin/application.py", line 495, in create
appshots-fastapi-myappshots-1  |     Form = await model_view.scaffold_form()
appshots-fastapi-myappshots-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/sqladmin/models.py", line 1046, in scaffold_form
appshots-fastapi-myappshots-1  |     return await get_model_form(
appshots-fastapi-myappshots-1  |            ^^^^^^^^^^^^^^^^^^^^^
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/sqladmin/forms.py", line 628, in get_model_form
appshots-fastapi-myappshots-1  |     field = await converter.convert(
appshots-fastapi-myappshots-1  |             ^^^^^^^^^^^^^^^^^^^^^^^^
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/sqladmin/forms.py", line 291, in convert
appshots-fastapi-myappshots-1  |     kwargs = await self._prepare_kwargs(
appshots-fastapi-myappshots-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/sqladmin/forms.py", line 154, in _prepare_kwargs
appshots-fastapi-myappshots-1  |     kwargs = await self._prepare_relationship(
appshots-fastapi-myappshots-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/sqladmin/forms.py", line 215, in _prepare_relationship
appshots-fastapi-myappshots-1  |     kwargs.setdefault("data", await self._prepare_select_options(prop, engine))
appshots-fastapi-myappshots-1  |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/sqladmin/forms.py", line 232, in _prepare_select_options
appshots-fastapi-myappshots-1  |     for obj in objects.scalars().all()
appshots-fastapi-myappshots-1  |                ^^^^^^^^^^^^^^^^^^^^^^^
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/result.py", line 1784, in all
appshots-fastapi-myappshots-1  |     return self._allrows()
appshots-fastapi-myappshots-1  |            ^^^^^^^^^^^^^^^
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/result.py", line 571, in _allrows
appshots-fastapi-myappshots-1  |     for made_row, sig_row in [
appshots-fastapi-myappshots-1  |                              ^
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/result.py", line 574, in <listcomp>
appshots-fastapi-myappshots-1  |     strategy(made_row) if strategy else made_row,
appshots-fastapi-myappshots-1  |     ^^^^^^^^^^^^^^^^^^
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/loading.py", line 257, in require_unique
appshots-fastapi-myappshots-1  |     raise sa_exc.InvalidRequestError(
appshots-fastapi-myappshots-1  | sqlalchemy.exc.InvalidRequestError: The unique() method must be invoked on this Result, as it contains results that include joined eager loads against collections

### Environment

- python:3.11.1-slim (Docker)
- sqladmin 0.12.0

**Comment by aminalaee** [comment]
That's very strange, checking the code there's nothing unusual happening.
Can you check this query with your current code?

```python
from sqlalchemy import select


result = session.execute(select(User))
print(result.scalars().all())
```

I'm guessing this should fail and it shouldn't be related to SQLAdmin.

**Comment by florianabel** [comment]
Hi @aminalaee, thanks a lot for getting back to me so quickly.

I tested the code snippet you provided and as you expected, it failed with the same error. (See stack trace below)

When I replace .all() with .unique(), it runs without error and provides me with the desired results.

Researching about the problem I found this and think it might be connected:

> The error message "sqlalchemy.exc.invalidrequesterror: the unique() method must be invoked on this result , as it contains r

> esults that include joined eager loads against collections" typically occurs when you are using a joined eager loading technique to load a relationship collection and you haven't included the unique() method in your query .
> 
> When using joined eager loading, the Query object must de-duplicate the rows against the leading entity. This is because the JOIN clause in the SQL query can cause rows to be duplicated when there are multiple child rows. The unique() method is used to remove these duplicates.
> 
> To resolve the error, you need to make sure that you include the unique() method in your query when using joined eager loading.
> 
> You can refer to the SQLAlchemy documentation on Relationship Loading Techniques for more information on using joined eager loading and the unique() method.

appshots-fastapi-myappshots-1  | INFO:     172.28.0.1:36648 - "GET /test HTTP/1.1" 500 Internal Server Error
appshots-fastapi-myappshots-1  | ERROR:    Exception in ASGI application
appshots-fastapi-myappshots-1  | Traceback (most recent call last):
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/protocols/http/httptools_impl.py", line 435, in run_asgi
appshots-fastapi-myappshots-1  |     result = await app(  # type: ignore[func-returns-value]
appshots-fastapi-myappshots-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
appshots-fastapi-myappshots-1  |     return await self.app(scope, receive, send)
appshots-fastapi-myappshots-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/applications.py", line 276, in __call__
appshots-fastapi-myappshots-1  |     await super().__call__(scope, receive, send)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/applications.py", line 122, in __call__
appshots-fastapi-myappshots-1  |     await self.middleware_stack(scope, receive, send)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 184, in __call__
appshots-fastapi-myappshots-1  |     raise exc
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 162, in __call__
appshots-fastapi-myappshots-1  |     await self.app(scope, receive, _send)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/sessions.py", line 86, in __call__
appshots-fastapi-myappshots-1  |     await self.app(scope, receive, send_wrapper)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/cors.py", line 83, in __call__
appshots-fastapi-myappshots-1  |     await self.app(scope, receive, send)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
appshots-fastapi-myappshots-1  |     raise exc
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
appshots-fastapi-myappshots-1  |     await self.app(scope, receive, sender)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/contextlib.py", line 222, in __aexit__
appshots-fastapi-myappshots-1  |     await self.gen.athrow(typ, value, traceback)
appshots-fastapi-myappshots-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/concurrency.py", line 36, in contextmanager_in_threadpool
appshots-fastapi-myappshots-1  |     raise e
appshots-fastapi-myappshots-1  | sqlalchemy.exc.InvalidRequestError: The unique() method must be invoked on this Result, as it contains results that include joined eager loads against collections

**Comment by florianabel** [comment]
Upon further testing I found that `lazy=joined` in the oauth_accounts relationship seems to be responsible for the error in this case. Removing it, also removes the error when using the sqladmin panel. Unfortunately, it seems to be necessary for the fastapi-users package to work correctly.

`class User(SQLAlchemyBaseUserTable[int], Base):
    id: int = Column(Integer, primary_key=True)
    oauth_accounts: Mapped[List[OAuthAccount]] = relationship(
        "OAuthAccount", lazy="joined"
    )`
    
   Based on this I am wondering if there is a "bug" in fastapi-users or if adding the unique() method would prove beneficial for sqladmin.

**Comment by aminalaee** [comment]
I don't think this is the `lazy="joined"`. I have tested SQLAlchemy models with `lazy="joined"` in SQLAdmin before.
I think `session.execute(select(User)).scalars().all()` is failing because internally FastAPI-Users is using eager loading, it's not in your code. I don't know much about fastapi-users but since this `unique()` filter is not in the database level and is only applied to the results, I think we can add it.

Feel free to do the PR.

**Comment by florianabel** [comment]
I just checked it again and you are right, it is not the `lazy="joined"`. 

Happy to help with the PR, I have already started on it and updated the discussed lines locally. 
This being my first contribution to a package I was wondering if you'd see any further test coverage needed here?
I wasn't sure about it.

Also, I am receiving errors on running `hatch lint` and `hatch check` , whereas `hatch test` runs without producing output.

`/bin/sh: lint: command not found`
`/bin/sh: check: command not found` 

Sorry, if this is the wrong place to clarify these details.
